blscosi/bdnproto/struct.go

```diff
@@ -0,0 +1,67 @@
+package bdnproto
+
+import (
+	"errors"
+	"fmt"
+
+	"go.dedis.ch/cothority/v3/blscosi/protocol"
+	"go.dedis.ch/kyber/v3"
+	"go.dedis.ch/kyber/v3/pairing"
+	"go.dedis.ch/kyber/v3/sign"
+	"go.dedis.ch/kyber/v3/sign/bdn"
+	"go.dedis.ch/kyber/v3/sign/cosi"
+)
+
+// BdnSignature is a signature that must be verified using coefficients
+// derived from the public keys
+type BdnSignature []byte
+
+// GetMask returns the bytes representing the mask
+func (sig BdnSignature) GetMask(suite pairing.Suite, pubkeys []kyber.Point) (*sign.Mask, error) {
+	return protocol.BlsSignature(sig).GetMask(suite, pubkeys)
+}
+
+// Point returns the point associated with the signature
+func (sig BdnSignature) Point(suite pairing.Suite) (kyber.Point, error) {
+	return protocol.BlsSignature(sig).Point(suite)
+}
+
+// Verify returns an error if the signature can't be verified or nil if it matches
+func (sig BdnSignature) Verify(suite pairing.Suite, msg []byte, pubkeys []kyber.Point) error {
+	policy := cosi.NewThresholdPolicy(protocol.DefaultThreshold(len(pubkeys)))
+
+	return sig.VerifyWithPolicy(suite, msg, pubkeys, policy)
+}
+
+// VerifyWithPolicy checks that the signature is correct and that the number of signers
+// matches the policy
+func (sig BdnSignature) VerifyWithPolicy(suite pairing.Suite, msg []byte, pubkeys []kyber.Point, policy cosi.Policy) error {
```

Comment:
I wonder if cosi.Policy still makes sense here, or if that type should be someplace else. Is this our only dependency on package cosi now? https://go-proverbs.github.io -> "A little copying is better than a little dependency."

Reply:
I agree it could/should be in the same package as the mask. Now if we move entirely, changes will be required for previous cosi packages. I guess we can do what you suggest a copy it to the new location with a deprecation notice ?